assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/ConfigureScreensPage.tsx

```diff
@@ -190,25 +193,16 @@ const ConfigurePlaceCard: ComponentType<ConfigurePlaceCardProps> = ({
 
   useEffect(() => {
     setPlacesAndScreensToUpdate((placesAndScreens) => {
-      const screensAtPlace = placesAndScreens[place.id];
-      const newState = { ...placesAndScreens };
-
-      if (screensAtPlace) {
-        newState[place.id].updated_pending_screens = updatedPendingScreens;
-        newState[place.id].new_pending_screens = newScreens;
-        newState[place.id].existing_pending_screens = existingScreensToArray(
-          existingPendingScreens
-        );
-      } else {
-        newState[place.id] = {
+      return {
+        ...placesAndScreens,
+        [place.id]: {
           updated_pending_screens: updatedPendingScreens,
           new_pending_screens: newScreens,
```

Comment:
Are there any other fields on `placesAndScreens[place.id]` that need to be preserved? In the updated state, this replaces `placesAndScreens[place.id]` with a new object with keys `updated_pending_screens`, `new_pending_screens`, `existing_pending_screens`.

If any existing fields besides those need to be preserved, then this should be
```suggestion
        [place.id]: {
          ...placesAndScreens[place.id],
          updated_pending_screens: updatedPendingScreens,
          new_pending_screens: newScreens,
```